Source/WebCore/css/parser/CSSTokenizerInputStream.cpp

```diff
@@ -45,11 +46,31 @@ void CSSTokenizerInputStream::advanceUntilNonWhitespace()
     // Using ASCII whitespace here rather than CSS space since we don't do preprocessing
     if (m_string->is8Bit()) {
         auto characters = m_string->span8();
-        while (m_offset < m_stringLength && isASCIIWhitespace(characters[m_offset]))
+        while (m_offset < m_stringLength && isASCIIWhitespace(characters[m_offset])) {
+            if (isNewline(characters[m_offset]))
+                return;
             ++m_offset;
+        }
     } else {
         auto characters = m_string->span16();
-        while (m_offset < m_stringLength && isASCIIWhitespace(characters[m_offset]))
+        while (m_offset < m_stringLength && isASCIIWhitespace(characters[m_offset])) {
+            if (isNewline(characters[m_offset]))
+                return;
+            ++m_offset;
+        }
+    }
+}
+
+void CSSTokenizerInputStream::advanceUntilNonWhitespaceOrNewline()
+{
+    // Using ASCII whitespace here rather than CSS space since we don't do preprocessing
+    if (m_string->is8Bit()) {
+        auto characters = m_string->span8();
+        while (m_offset < m_stringLength && (isASCIIWhitespace(characters[m_offset]) || isNewline(characters[m_offset])))
+            ++m_offset;
+    } else {
+        auto characters = m_string->span16();
+        while (m_offset < m_stringLength && (isASCIIWhitespace(characters[m_offset]) || isNewline(characters[m_offset])))
             ++m_offset;
     }
 }
```

Comment:
The `isASCIIWhitespace` function already returns `true` for every character that `isNewline` returns `true` for, so there is no need to add the `isNewline` check here.

Reply:
Fixed. I originally added in the `|| isNewline(characters[m_offset])` to make it more clear in the code, but was unnecessary.